rewrite-java/src/main/java/org/openrewrite/java/trait/VariableAccess.java

```diff
@@ -0,0 +1,193 @@
+     */
+    public boolean isWriteAccess() {
+        MaybeParenthesesPair pair = MaybeParenthesesPair.from(cursor);
+        if (pair.getParent() instanceof J.Assignment) {
+            J.Assignment assignment = (J.Assignment) pair.getParent();
+            return assignment.getVariable() == pair.getTree();
+        }
+        if (pair.getParent() instanceof J.Unary) {
+            J.Unary unary = (J.Unary) pair.getParent();
+            return unary.getExpression() == pair.getTree();
+        }
+        return false;
+    }
+
+    /**
+     * A read access to a variable. In other words, it is a variable access that does
+     * _not_ occur as the destination of a simple assignment, but it may occur as
+     * the destination of a compound assignment or a unary assignment.
+     */
+    public boolean isReadAccess() {
+        if (isWriteAccess()) {
+            return false;
+        }
+        MaybeParenthesesPair pair = MaybeParenthesesPair.from(cursor);
+        if (pair.getParent() instanceof J.Assignment) {
+            J.Assignment assignment = (J.Assignment) pair.getParent();
+            return assignment.getVariable() != pair.getTree();
+        }
+        return true;
+    }
+
+    public static class Matcher extends SimpleTraitMatcher<VariableAccess> {
+        private Predicate<JavaType> typeTest = t -> true;
+
+        public Matcher isOfType(JavaType type) {
+            this.typeTest = t -> TypeUtils.isOfType(t, type);
+            return this;
+        }
+
+        public Matcher isOfClassType(String fullyQualifiedName) {
+            this.typeTest = t -> TypeUtils.isOfClassType(t, fullyQualifiedName);
+            return this;
+        }
+
+        @Override
+        public <P> TreeVisitor<? extends Tree, P> asVisitor(VisitFunction3<VariableAccess, P> visitor) {
+            return new JavaVisitor<P>() {
+                @Override
+                public J visitIdentifier(J.Identifier ident, P p) {
+                    VariableAccess varAccess = test(getCursor());
+                    return varAccess != null ?
+                            (J) visitor.visit(varAccess, getCursor(), p) :
+                            super.visitIdentifier(ident, p);
+                }
+            };
+        }
+
+        /**
+         * Since variable access is a trait of a leaf LST element (J.Identifier),
+         * there are no higher elements in the tree that could potentially match.
+         */
+        @Override
+        public Stream<VariableAccess> higher(Cursor cursor) {
+            return Stream.empty();
+        }
+
+        @Nullable
+        @Override
+        protected VariableAccess test(Cursor cursor) {
+            if (!(cursor.getValue() instanceof J.Identifier)) {
+                return null;
+            }
+
+            J.Identifier ident = cursor.getValue();
+            String name = ident.getSimpleName();
+            if (name.equals("this") || name.equals("super") ||
```

Comment:
This is Java specific. Do you care at this point?

Reply:
Not yet. I saw the same in rewrite-analysis.